surfactant/cmd/cli.py

```diff
@@ -52,25 +55,142 @@ def find(sbom, output_format, input_format, **kwargs):
 @click.command("edit")
-def edit(sbom):
+def edit(sbom, output_format, input_format, **kwargs):
     "CLI command to edit specific entry(s) in a supplied SBOM"
 
 
-@click.argument("sbom", type=click.File("r"), required=True)
-@click.command("add")
-def add(sbom):
-    "CLI command to add specific entry(s) to a supplied SBOM"
+class cli_add:
+    """
+    A class that implements the surfactant cli add functionality
+
+    Attributes:
+    match_functions         A dictionary of functions that provide matching functionality for given SBOM fields (i.e. uuid, sha256, installpath, etc)
+    camel_case_conversions  A dictionary of string conversions from all lowercase to camelcase. Used to convert python click options to match the SBOM attribute's case
+    sbom                    An internal record of sbom entries the class adds to as it finds more matches.
+    """
+
+    camel_case_conversions: dict
+    match_functions: dict
+    sbom: SBOM
+
+    def __init__(self):
+        """Initializes the cli_add class"""
+        self.match_functions = {
+            "relationship": self.add_relationship,
+            "file": self.add_file,
+            "installPath": self.add_installpath,
+            "entry": self.add_entry,
+        }
+        self.camel_case_conversions = {
+            "uuid": "UUID",
+            "filename": "fileName",
+            "installpath": "installPath",
+            "capturetime": "captureTime",
+            "relationshipassertion": "relationshipAssertion",
+        }
+
+    def handle_kwargs(self, kwargs: dict) -> dict:
+        converted_kwargs = {}
+        for k, v in kwargs.items():  # Convert key values to camelcase where appropriate
+            key = self.camel_case_conversions[k] if k in self.camel_case_conversions else k
+            converted_kwargs[key] = v
+        return converted_kwargs
+
+    def execute(self, input_sbom: SBOM, **kwargs):
+        """Executes the main functionality of the cli_find class
+        param: input_sbom   The sbom to add entries to
+        param: kwargs:      Dictionary of key/value pairs indicating what features to match on
+        """
+        converted_kwargs = self.handle_kwargs(kwargs)
+        self.sbom = input_sbom
+
+        for key, value in converted_kwargs.items():
+            if key in self.match_functions:
+                self.match_functions[key](value)
+            else:
+                logger.warning(f"Paramter {key} is not supported")
+        return self.sbom
+
+    def add_relationship(self, value: dict) -> bool:
+        self.sbom.add_relationship(Relationship(**value))
+
+    def add_file(self, path):
+        self.sbom.software.append(Software.create_software_from_file(path))
+
+    def add_entry(self, entry):
+        self.sbom.software.append(Software.from_dict(entry))
+
+    def add_installpath(self, prefixes: tuple):
+        cleaned_prefixes = tuple([p.rstrip("/") for p in prefixes])
+        containerPathPrefix, installPathPrefix = cleaned_prefixes
```

Comment:
pre-commit thinks a generator could be used here... if we stick with a tuple, this could be changed to `cleaned_prefixes = tuple(cleaned_prefixes = tuple(p.rstrip("/") for p in prefixes)` to get rid of the list comprehension that it is probably complaining about.

The generator approach would skip the packing into a tuple then unpacking and end up as something like:
```python
cleaned_prefixes = (p.rstrip("/") for p in prefixes)
containerPathPrefix, installPathPrefix = cleaned_prefixes
```

Reply:
Ah I keep forgetting to check the pre-commit hooks. Ive updated with your generator suggestion